networks/rpc/http.go

```diff
@@ -48,9 +49,12 @@ var nullAddr, _ = net.ResolveTCPAddr("tcp", "127.0.0.1:0")
 
 type httpConn struct {
 	client    *http.Client
+	url       string
 	req       *http.Request
 	closeOnce sync.Once
 	closed    chan struct{}
+	mu        sync.Mutex
```

Comment:
```suggestion
	mu        sync.Mutex // protects headers
```

It might be useful for later to write down the use of the mutex.

Reply:
I added the comment. Thanks!